src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -1072,17 +1077,24 @@ Copyright (c) .NET Foundation. All rights reserved.
       $(PublishItemsOutputGroupDependsOn);
       ResolveReferences;
       ComputeFilesToPublish;
+      BundlePublishDirectory;
     </PublishItemsOutputGroupDependsOn>
   </PropertyGroup>
 
   <Target Name="PublishItemsOutputGroup" DependsOnTargets="$(PublishItemsOutputGroupDependsOn)" Returns="@(PublishItemsOutputGroupOutputs)">
     <ItemGroup>
-      <PublishItemsOutputGroupOutputs Condition="'$(GenerateDependencyFile)' == 'true' and '$(_UseBuildDependencyFile)' != 'true'"
+      <PublishItemsOutputGroupOutputs Condition="'$(GenerateDependencyFile)' == 'true' and '$(_UseBuildDependencyFile)' != 'true' and '$(PublishSingleFile)' != 'true'"
                                       Include="$(PublishDepsFilePath)"
                                       TargetPath="$(ProjectDepsFileName)" />
       <PublishItemsOutputGroupOutputs Include="@(ResolvedFileToPublish->'%(FullPath)')"
                                       TargetPath="%(ResolvedFileToPublish.RelativePath)"
                                       IsKeyOutput="%(ResolvedFileToPublish.IsKeyOutput)"
+                                      Condition="'$(PublishSingleFile)' != 'true' or '%(ResolvedFileToPublish.ExcludeFromSingleFile)'=='true'"
+                                      OutputGroup="PublishItemsOutputGroup" />
+      <PublishItemsOutputGroupOutputs Include="$(PublishedSingleFilePath)"
```

Comment:
Why is this step necessary? 
The AppHost executable should already be in `ResolvedFileToPublish`, added by  `_ComputeCopyToPublishDirectoryItems`. 
The AppHost name and the single-file name are the same. So, it shouldn't be necessary to add the singlefile path explicitly.

Reply:
The bundled version of the exe is never added to the ResolvedFileToPublish (or any) item group.  The non-bundled version that is in the item group will now get intentionally excluded in the single file case and the bundled version will be included instead.

Reply:
OK the AppHost itself is not marked `ExcludeFromSingleFile`, but handled implicitly; hence the extra explicit copy. Thanks.